sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientBuilderTest.java

```diff
@@ -84,6 +89,37 @@ public void trainingClientBuilderDefaultPipeline(HttpClient httpClient, FormReco
             assertNotNull(clientBuilder.buildClient().getFormRecognizerClient().beginRecognizeContentFromUrl(input).getFinalResult()));
     }
 
+    /**
+     * Test for invalid endpoint.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void trainingClientBuilderInvalidEndpoint(HttpClient httpClient, FormRecognizerServiceVersion serviceVersion) {
+        clientBuilderWithDefaultPipelineRunner(httpClient, serviceVersion, clientBuilder -> (input) -> {
+            Exception exception =  assertThrows(RuntimeException.class, () ->
+                clientBuilder.endpoint(INVALID_ENDPOINT).buildClient().getFormRecognizerClient()
+                    .beginRecognizeContentFromUrl(input).getFinalResult());
+            // "Max retries 3 times exceeded. Error Details: Connection refused: no further information:
+            // notreal.azure.com/23.217.138.110:443"
+            // Recording mode has Error Details: null
+            assertTrue(exception.getMessage().contains("Max retries 3 times exceeded. Error Details:"));
+        });
+    }
+
+    /**
+     * Test for http endpoint, which throws HTTPS requirement exception message.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void trainingClientBuilderHttpEndpoint(HttpClient httpClient, FormRecognizerServiceVersion serviceVersion) {
```

Comment:
not  sure, if the test name is clear.

Reply:
this test renamed to httpsProtocolRequiredException and moved to TrainingClientBuilderUniteTest